### ISSUE TITLE
Ability to review/modify email templates

### ISSUE THREAD

**Issue by saralouhicks** [reactioncommerce/reaction#423]
The following email templates are needed for Beta:

New user invite (admin)
Welcome (shopper)
Order complete (shopper)
Shipment notification (shopper)
Order status changes - refunds, delays, stage change (shopper)
Change PW (shopper)

**Comment by aaronjudd** [comment]
@mikemurray @saralouhicks per discussed, this is the list of email template we need/have.

We currently don't send the order complete or welcome emails, the others are sent, but we need a good standard design template to use with these.

**Comment by aaronjudd** [comment]
Templates will also need i18n handling. The current ones need updates as well for i18n in the sender fields on the server. https://github.com/reactioncommerce/reaction/blob/ca3e454fe6bd9a3998e15dce444992be20803229/packages/reaction-core/server/methods/orders.js#L197

**Comment by aaronjudd** [comment]
@saralouhicks we'll need some generic templates to add to the existing templates. (which have no layout).

**Comment by prinzdezibel** [comment]
i18n is addressed with PR https://github.com/reactioncommerce/reaction/pull/1062

**Comment by zenweasel** [comment]
So since this ticket was assigned to @rymorgan I am assuming this is the epic and covers doing the design UI of the templates in anticipation of implementation starting for 16?

**Comment by rymorgan** [comment]
Here's the list of transaction emails with copy written for the first round of emails we want to include: https://docs.google.com/a/reactioncommerce.com/spreadsheets/d/1MlHv3gqk3EjpQsan3-qC5I4FN-CEp-JjamiUnit7tMY/edit?usp=sharing

Two designs have been sent for initial coding.  The coded emails will need to be vetted by @kieckhafer or @mikemurray. We will also need to propagate the code to accommodate the rest of the emails which will mostly amount to text edits.

**Comment by kieckhafer** [comment]
This is a WIP. Work on #1318 needs to be completed prior to work on this ticket being dove into.

**Comment by HadenHiles** [comment]
Just wondering if someone can point me in the right direction as to how to override the order email templates using the Templates collection? I can't seem to find anything in the docs or otherwise.

Thanks!

**Comment by aaronjudd** [comment]
@HadenHiles currently you can override by adding the html email content directly into the `Templates` collection.

**Comment by HadenHiles** [comment]
I noticed that in the `email.js`. I'm just not sure as to what the `Templates` collection should look like. Could you provide an example template object perhaps? I am branched off the development branch and there is no `Templates` collection with example data for me to reference, and I couldn't find any examples in the docs either.

Sorry if I missed something that already explains this with more detail, I wouldn't ask without doing my best to figure it out on my own. I even tried guessing the object structure but had no luck.

**Comment by jshimko** [comment]
You can find the schema here:

https://github.com/reactioncommerce/reaction/blob/master/lib/collections/schemas/templates.js

So, to replace the welcome email, it might look something like:

```js
import Templates from '/lib/collections';

Templates.insert({
  template: 'accounts/sendWelcomeEmail',
  language: 'en',
  source: 'the HTML or plain text content of your email'
});
```

**Comment by HadenHiles** [comment]
Ah, thank you @jshimko!

**Comment by aaronjudd** [comment]
The current `Templates` schema is https://github.com/reactioncommerce/reaction/blob/master/lib/collections/schemas/templates.js

We didn't really document that because we're also going to be updating it soon with a more flexible  structure, see the proposed structure here: https://github.com/reactioncommerce/reaction-docs/compare/proposed-email-templates

**Comment by HadenHiles** [comment]
Oh I see; why document something that is going to be updated and redundant. That makes sense.

Thanks again for the help, seems like you guys are really on your game. Keep it up!

**Comment by zenweasel** [comment]
@saralouhicks @aaronjudd  So the title of this ticket is about reviewing/editing the email templates however the body of the ticket is all about implementing the templates which is now in #1429. Do we want to keep this ticket for adding the ability to edit templates through the dashboard or start over with a new ticket?

**Comment by zenweasel** [comment]
Since this ticket mostly addresses email templates which is covered in #1429 I am going to close this ticket. A new ticket for modifying/editing tickets through the dashboard should be created.

**Comment by zenweasel** [comment]
This is the transactional email "epic". Reopening

**Comment by kieckhafer** [comment]
Fixed by #1585
